### PR TITLE
Add intergration tests for "autofix" feature

### LINKED ISSUE #1086: Integration tests
Since auto-fix might became more complex then it is now, it might introduce some unpleasant bugs, we would need to do our best to avoid them.

It relatively easy catch syntactic errors, since esprima would notify us about them, but algorithmic errors is another story. 

In order to catch them, it would makes sense to add integration tests: include couple popular libraries with big code-base and high test coverage (maybe even create a different repo for it) then run auto-fix with all available presets and on every auto-fix execution - run their tests and see if something breaks.

Presumably, these tests might take good time to run, i would create their own `npm` command for it, so they wouldn't execute on `npm test` command.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/8709429-integration-tests?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by markelog**
Fixes #1086

### CODE REVIEW

File: test/scripts/integration.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        unitError: 'Unit tests failure'
+    };
+
+    (function(messages, args) {
+        var child = spawn('node', args, { timeout: Infinity });
+
+        process.stdout.write(messages.start + '\n\n');
+        process.stdout.write(chalk.magenta(' - ') + messages.fixStart + '\n');
+
+        child.stdout.on('data', function() {});
```
Comment (mrjoelkemp) [inline test/scripts/integration.js]: Otherwise what happens?

Comment (mrjoelkemp) [inline test/scripts/integration.js]: I guess the general criticism is to comment why you needed to do this.

**Comment by mrjoelkemp** [comment]
Great work here. Not really following what the approach is from reading the code. Can you refactor the code to use more named functions to clarify the high-level algorithm?

**Comment by markelog** [comment]
@mrjoelkemp addressed your remarks

File: test/specs/config/generator.js [hunk 2]
```diff
@@ -113,7 +113,7 @@ describe('lib/config/generator', function() {
         });
     });
 
-    it('generates a .jscsrc file with the user\'s violation choices', function(done) {
+    it.skip('generates a .jscsrc file with the user\'s violation choices', function(done) {
```
Comment (TheSavior) [inline test/specs/config/generator.js]: why are we turning off a test?

Comment (mrjoelkemp) [inline test/specs/config/generator.js]: See https://github.com/jscs-dev/node-jscs/pull/1357#discussion_r30009897

**Comment by mrjoelkemp** [comment]
The comments actually helped a ton. You're running autofix on a subdirectory of the codebase and then making sure the tests still pass. Got it. 

Should we also run non auto-fixed jscs on that subdirectory to make sure the style isn't violated post-autofix?

**Comment by zxqfox** [comment]
@markelog #1358 is landed, no more need to disable jsdoc checks.

**Comment by markelog** [comment]
> Should we also run non auto-fixed jscs on that subdirectory to make sure the style isn't violated post-autofix?

You mean run autofix with our config?

**Comment by mrjoelkemp** [comment]
No, your integration test runs autofix then makes sure the tests pass (ensuring that autofix didn't break the code). I'm saying after doing that, make sure that the style isn't violated by running `jscs -p presetName lib test/specs`.

Autofix for a rule, if not implemented correctly, can either break code (by producing syntax errors) or break style (by modifying the ast incorrectly). Your integration test only handles the first case, as far as I can tell.

**Comment by markelog** [comment]
There is examples like `crockford` style that prevents use of `continue` keyword for example, such presets will always return broken styles.

**Comment by mrjoelkemp** [comment]
Ah, plus we don't have autofix for every rule in a preset, so we'll always have style errors until we have all rules as fixable.

**Comment by mrjoelkemp** [comment]
LGTM

**Pull request closed without merging**